### PR TITLE
Fix product weight when changing combination

### PR DESCRIPTION
**Pull request by fatmaBouchekoua**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | The weight of combination is calculated incorrectly.
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-1899
| How to test?  | Go to product detail page in FO and change combination, the weight will change according to the combination's impact of weight.

See: https://github.com/PrestaShop/PrestaShop/pull/7105

### CODE REVIEW

File: src/Core/Product/ProductPresenter.php [hunk 1]
```diff
@@ -488,6 +488,40 @@ public function addReferenceToDisplay(array $product, array $presentedProduct)
...
+    /**
+     * @param array $product
+     * @return int
+     */
+    public function getProductWeight(array $product){
+        $product_weight = 0;
+        if (isset($product['features'] )) {
+            foreach ($product['features'] as $feature) {
+                if ( '4' === $feature['id_feature']){
+                    $product_weight = $feature['value'];
```
Comment (aleeks) [inline src/Core/Product/ProductPresenter.php]: an int casting ? (int) $feature['value'];

File: src/Core/Product/ProductPresenter.php [hunk 2]
```diff
@@ -488,6 +488,40 @@ public function addReferenceToDisplay(array $product, array $presentedProduct)
...
     /**
+     * @param array $product
+     * @param array $presentedProduct
+     * @return array
+     */
+    public function AddWeightToDisplay(array $product, array $presentedProduct){
+        $product_weight = $this->getProductWeight($product);
+        foreach ($product['attributes'] as $attribute) {
+            if (isset($attribute['weight']) && 0 !== $attribute['weight']) {
+                $presentedProduct['weight_to_display'] = $attribute['weight'] + $product_weight;
```
Comment (aleeks) [inline src/Core/Product/ProductPresenter.php]: (int) $attribute['weight'] ?

Comment (mickaelandrieu) [inline src/Core/Product/ProductPresenter.php]: weight is a float, isn't it ?

Comment (aleeks) [inline src/Core/Product/ProductPresenter.php]: Yes! need to be casted. thank @mickaelandrieu

**Comment by aleeks** [comment]
@fatmaBouchekoua Hello,
need to cast, rebase, & report on develop :p !

File: src/Core/Product/ProductPresenter.php (hunk 1 above)
Comment (kpodemski) [inline src/Core/Product/ProductPresenter.php]: hardcoded id? where these $features comes from? these are from PREFIX_features?

File: themes/classic/templates/catalog/_partials/product-details.tpl [hunk 3]
```diff
@@ -50,7 +50,11 @@
           <dl class="data-sheet">
             {foreach from=$product.features item=feature}
               <dt class="name">{$feature.name}</dt>
-              <dd class="value">{$feature.value}</dd>
+              {if $feature.id_feature === '4'}
```
Comment (aleeks) [inline themes/classic/templates/catalog/_partials/product-details.tpl]: .. :( we don't have any other way to do that isn't ?

**Comment by fatmaBouchekoua** [comment]
@aleeks done!
Thanks..

File: themes/classic/templates/catalog/_partials/product-details.tpl (hunk 3 above)
Comment (kpodemski) [inline themes/classic/templates/catalog/_partials/product-details.tpl]: Sorry guys but there's always a choice, you can't hardcode id of feature when this feature is most of the time removed by `pscleaner` module after first installation of PrestaShop

Comment (kpodemski) [inline themes/classic/templates/catalog/_partials/product-details.tpl]: I'm little lost, what is the point of using weight from the "features" in the first place? You have `weight` field, or you should in product information/product option tab. Features are only for informations, not for calculations.

Comment (aleeks) [inline themes/classic/templates/catalog/_partials/product-details.tpl]: I'm totally agree with you.. some merchant wants this functionality apparently  (like price or discount..) http://forge.prestashop.com/browse/BOOM-1899

Comment (kpodemski) [inline themes/classic/templates/catalog/_partials/product-details.tpl]: @aleeks it's ok to change the "weight" and other informations about product related to the stuff inserted into "Combinations" tab but "Features" are completely different part of the system.

Comment (aleeks) [inline themes/classic/templates/catalog/_partials/product-details.tpl]: Yes, agree. We will do this with weight product (on shipping), not feature. It makes more sense isn't ?

Comment (kpodemski) [inline themes/classic/templates/catalog/_partials/product-details.tpl]: Yes, it does, it's important to remember that features are only for information and shouldn't be dynamic, unless we'll have the feature to... have features related to combinations :)

Comment (fatmaBouchekoua) [inline themes/classic/templates/catalog/_partials/product-details.tpl]: I agree. It does make sense ! i'll fix it.
Thanks!

**Comment by kpodemski** [comment]
@vTerenti how can you approve it? 😱

**Comment by aleeks** [comment]
Hello, so.

- @kpodemski yes, @vTerenti  approuves only because it was working :p
- See with @vincentbz, It should not work like that. 
- We will consider applying the features to the declensions, but not mix features/shipping. This, is not a bug fix. Thank you @kpodemski for your warning!

**Pull request closed without merging**